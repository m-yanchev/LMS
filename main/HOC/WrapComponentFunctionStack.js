function WrapComponentFunctionStack ( WrapComponent, HOCData ) {
    try {
        return HOCData.reduce( ( prevComponent, item ) => {
            return item.name(prevComponent, item.otherProps)
        }, WrapComponent );
    } catch (e) {
        console.error(e)
    }
}

export default WrapComponentFunctionStack