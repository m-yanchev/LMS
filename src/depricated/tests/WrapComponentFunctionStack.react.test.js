import React from 'react'
import WrapComponentFunctionStack from '../../client/entryPoints/main/HOC/WrapComponentFunctionStack'
import renderer from "react-test-renderer";

test('functionStack', () => {

    const funcData = [
        {
            name: func1,
            otherProps: {value:2, data:{name: 'Nicholas', phone: '+7 917 951 75 34' }}
        },{
            name: func2
        },{
            name: func3,
            otherProps: {}
        },{
            name: func4,
            otherProps: {str:'yes'}
        },{
            name: func5,
            otherProps: {array:[1, 2, 3]}
        }];

    class ReactComponent extends React.Component {
        render() {
            return <div>Hello</div>
        }
    }

    const ResultComponent = WrapComponentFunctionStack( ReactComponent, funcData);

    const component = renderer.create(
        <ResultComponent/>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    function func1(WrapComponent, otherProps) {
        return class extends React.Component {
            render(){
                return (<>
                        <WrapComponent/>
                        <p>{otherProps.value}</p>
                        <p>{otherProps.data.name}</p>
                        <p>{otherProps.data.phone}</p>
                    </>
                )
            }
        }
    }

    function func2(WrapComponent) {
        return class extends React.Component {
            render(){
                return (<>
                        <WrapComponent/>
                        <p>func2</p>
                    </>
                )
            }
        }
    }

    function func3(WrapComponent) {
        return class extends React.Component {
            render(){
                return (<>
                        <WrapComponent/>
                        <p>func3</p>
                    </>
                )
            }
        }
    }

    function func4(WrapComponent, otherProps) {
        return class extends React.Component {
            render(){
                return (<>
                        <WrapComponent/>
                        <p>{otherProps.str}</p>
                    </>
                )
            }
        }
    }

    function func5(WrapComponent) {
        return class extends React.Component {
            render(){
                return <WrapComponent/>
            }
        }
    }
});