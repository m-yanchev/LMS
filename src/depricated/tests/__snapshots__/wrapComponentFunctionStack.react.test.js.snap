// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`functionStack 1`] = `
Array [
  <div>
    Hello
  </div>,
  <p>
    2
  </p>,
  <p>
    Nicholas
  </p>,
  <p>
    +7 917 951 75 34
  </p>,
  <p>
    func2
  </p>,
  <p>
    func3
  </p>,
  <p>
    yes
  </p>,
]
`;
